general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -774,3 +815,17 @@ def find_api_key() -> str:
         "Try passing an 'api_key' variable, or setting your API key in the command line "
         "with SUPERSTAQ_API_KEY=..."
     )
+
+
+@dataclass
+class TargetInfo:
+    """A class to store data returned from a /get_targets request."""
+
+    target: str
+    properties: Dict[str, bool]
```

Comment:
can we flatten this so instead of `properties` we have e.g. `TargetInfo.available`?

this also might be a really place to start using `pydantic` - instead of a dataclass, we could use use
```python
class TargetInfo(pydantic.BaseModel):
    """A class to store data returned from a /get_targets request."""

    target: str
    available: bool = False
    ...
```
so that we could instantiate it with e.g.
```python
TargetInfo(target=target, **properties)
```
(and it will automatically handle defaults/validation/etc for us)

Reply:
Like the idea of using `pydantic`, will update. As for the decision to use `properties`, it was because I thought that if more fields were added, we wouldn't need to create a new variable for each one in the original dataclass. I was thinking that the user wouldn't use the `TargetInfo` dataclass directly, but instead would be able to do `get_targets(available=True)` or any keyword supported in `properties`